storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -2570,10 +2570,16 @@ bool copy_back(int argc, char **argv) {
 
 bool decrypt_decompress_file(const char *filepath, uint thread_n) {
   std::stringstream cmd, message;
-  char *dest_filepath = strdup(filepath);
+  char buf[FN_LEN];
   bool needs_action = false;
 
-  cmd << "cat " << filepath;
+  if (escape_string_for_mysql(&my_charset_utf8mb4_general_ci, buf, 0, filepath,
+                              strlen(filepath)) == (size_t)-1) {
```

Comment:
I think the commit message doesn't talk about this change. Why do we use it? it is escaping the new line characters?
Can you please update commit message? and a comment here too

Reply:
Hi @satya-bodapati .
The fix is about escaping the filepath with single quotes voa SQUOTE, however, if the file name has quotes in it we can still break out the command and inject code. escape_string_for_mysql is meant to scape special characters (include quotes) within the file name.

I will adjust the 2.4 PR and re-import it to 8.0

Reply:
Thanks marce, please rerequest review after PR update